src/WP_CLI/CommandWithMeta.php

```diff
@@ -254,6 +254,127 @@ public function update( $args, $assoc_args ) {
 	}
 
 	/**
+	 * Get meta field value.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <id>
+	 * : The ID of the object.
+	 *
+	 * <key>
+	 * : The name of the meta field to get.
+	 *
+	 * <pluck-key>
+	 * : The name of the inner key to get.
+	 *
+	 * [--delimiter=<character>]
+	 * : The character to delimit hierarchy within the pluck-key.
+	 * ---
+	 * default: .
+	 * ---
+	 *
+	 * [--format=<format>]
+	 * : Accepted values: table, json. Default: table
+	 *
+	 */
+	public function pluck( $args, $assoc_args ) {
+		list( $object_id, $meta_key, $pluck_key ) = $args;
+
+		$object_id = $this->check_object_id( $object_id );
+
+		$value = get_metadata( $this->meta_type, $object_id, $meta_key, true );
+
+		$traverser = new RecursiveDataStructureTraverser( $value );
+		$traverser->set_delimiter( $assoc_args['delimiter'] );
+
+		try {
+			$value = $traverser->get( $pluck_key );
+		} catch ( \Exception $e ) {
+			die( 1 );
+		}
+
+		WP_CLI::print_value( $value, $assoc_args );
+	}
+
+	/**
+	 * Update a multi-dimensional meta field.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <id>
+	 * : The ID of the object.
+	 *
+	 * <key>
+	 * : The name of the meta field to update.
+	 *
+	 * <patch-key>
+	 * : The name of the inner key to update.
+	 *
+	 * [<value>]
+	 * : The new value. If omitted, the value is read from STDIN.
+	 *
+	 * [--delimiter=<character>]
+	 * : The character to delimit hierarchy within the patch-key.
+	 * ---
+	 * default: .
+	 * ---
+	 *
+	 * [--mode=<mode>]
+	 * : Configures the behavior of the value update.
```

Comment:
No test for this yet but it would unset the key which is different than say `wp post meta patch 1 meta-key foo.bar ''` and just setting it to an empty value.

I like the idea of using a separate command but can't think of a good name to use for that. I think that it can work under the umbrella of `patch` as it is still a nested value modification unless we can come up with a verb that makes more sense? It's "patching" the value with the key removed.

I was also thinking that there may be a `merge` mode a merge of data rather than just replacing, but I pulled back on it as it probably wouldn't be very practical.